assets/js/theme/common/carousel/utils/handleImageAspectRatio.js

```diff
@@ -0,0 +1,45 @@
+import getActiveSlideInfo from './getActiveSlideInfo';
+
+const IMAGE_CLASSES = {
+    vertical: 'is-vertical-image-type',
+    square: 'is-square-image-type',
+};
+const IS_ANALYZED_DATA_ATTR = 'image-ratio-analyzed';
+
+const defineClass = (imageAspectRatio) => {
+    switch (true) {
+    case imageAspectRatio > 0.8 && imageAspectRatio <= 1.2:
+        return IMAGE_CLASSES.square;
+    case imageAspectRatio > 1.2:
+        return IMAGE_CLASSES.vertical;
+    default:
+        return '';
+    }
+};
+
+const updateImages = ($slides, imageNode) => {
+    const imageAspectRatio = imageNode.naturalHeight / imageNode.naturalWidth;
```

Comment:
Why did we move from client values to natural?

Reply:
to define image aspect ratio and when image loaded, to know did it load successfully